pulse2percept/files.py

```diff
@@ -6,8 +6,453 @@
+    set_skvideo_path(ffmpeg_path, libav_path)
+
+    if skvideo._HAS_FFMPEG:
+        backend = 'ffmpeg'
+    else:
+        backend = 'libav'
+    video = svio.vread(filename, as_grey=as_gray, backend=backend)
+    logging.getLogger(__name__).info("Loaded video from file '%s'." % filename)
+    d_s = "Loaded video has shape (T, M, N, C) = " + str(video.shape)
+    logging.getLogger(__name__).debug(d_s)
+
+    if as_timeseries:
+        # TimeSeries has the time as the last dimensions: re-order dimensions,
+        # then squeeze out singleton dimensions
+        axes = np.roll(range(video.ndim), -1)
+        video = np.squeeze(np.transpose(video, axes=axes))
+        fps = load_video_framerate(filename)
+        d_s = "Reshaped video to shape (M, N, C, T) = " + str(video.shape)
+        logging.getLogger(__name__).debug(d_s)
+        return utils.TimeSeries(1.0 / fps, video)
+    else:
+        # Return as ndarray
+        return video
+
+
+def load_video_generator(filename, ffmpeg_path=None, libav_path=None):
+    """Returns a generator that can load a video from file frame-by-frame.
+
+    This function loads a video from file and returns it as a NumPy ndarray
+    with the help of the Scikit-Video.
+
+    Parameters
+    ----------
+    filename : str
+        Video file name
+    ffmpeg_path : str, optional
+        Path to ffmpeg library. If not given, uses the system's default path.
+    libav_path : str, optional
+        Path to libav library. If not given, uses the system's default path.
+
+    Returns
+    -------
+    reader : skvideo.io.FFmpegReader | skvideo.io.LibAVReader
+        A Scikit-Video reader object
+
+    Examples
+    --------
+    >>> from skvideo import datasets
+    >>> reader = load_video_generator(datasets.bikes())
+    >>> for frame in reader.nextFrame():
+    ...    pass
+
+    """
+    if not has_skvideo:
+        raise ImportError("You do not have scikit-video installed. "
+                          "You can install it via $ pip install sk-video.")
+
+    # Set the path if necessary
+    set_skvideo_path(ffmpeg_path, libav_path)
+
+    # Try loading
+    if skvideo._HAS_FFMPEG:
+        reader = svio.FFmpegReader(filename)
+    elif skvideo._HAS_AVCONV:
+        reader = svio.LibAVReader(filename)
+    else:
+        raise ImportError("You have neither ffmpeg nor libav (which comes "
+                          "with avprobe) installed.")
+
+    return reader
+
+
+def save_video(data, filename, width=None, height=None, fps=30,
+               ffmpeg_path=None, libav_path=None):
+    """Saves a video to file.
+
+    This function stores a NumPy ndarray to file using Scikit-Video.
+
+    Parameters
+    ----------
+    data : ndarray | pulse2percept.utils.TimeSeries
+        Video data as a NumPy ndarray must have dimension (T, M, N, C),
+        (T, M, N), (M, N, C), or (M, N), where T is the number of frames,
+        M is the height, N is the width, and C is the number of channels.
```

Comment:
I am using a [Scikit-Video utility](https://github.com/scikit-video/scikit-video/blob/master/skvideo/utils/__init__.py#L279) to interpret the input dimensions. The utility checks the last dimension: If it is small (<=4), it's most likely a color channel, hence the input is interpreted as (M, N, C). Else it is interpreted as (T, M, N).

I have updated the doc strings to make this clear.